content/blaze.md

```diff
@@ -0,0 +1,543 @@
+    instance.saveList();
+  }
+});
+```
+
+<h3 id="scope-dom-lookups-to-instance">Scope DOM lookups to the template instance</h3>
+It's a bad idea to look up things directly in the DOM with jQuery's global `$()`. It means you are relying on things rendered outside of your component not interacting with your selector. Also, it limits your options on rendering *outside* of the main document (see testing section below).
+
+Instead, Blaze gives you `instance.$()` which scopes a lookup to within the current template instance. Typically you use this either from a `onRendered()` callback to setup external (e.g. jQuery) plugins, or from event handlers to call direct DOM functions. For instance, when a user adds a new todo, we want to focus it's `<input>` element:
+
+```js
+Template.listsShow.events({
+  'click .js-todo-add'(event, instance) {
+    instance.$('.js-todo-new input').focus();
+  }
+});
+```
+
+<h3 id="js-selectors-for-events">Use `.js-` selectors for event maps</h3>
+When you are setting up event maps in your JS files, you need to 'select' the element in the template that the event attaches to. Rather than using the same CSS classnames that are used to style the elements, it's better practice to use classnames that are specifically added for those event maps. A reasonable convention is a class starting with `js-` to indicate it is used by the JavaScript. For instance `.js-todo-add` above.
+
+<h3 id="passing-template-content">Passing HTML content as template arguments</h3>
+If you need to pass in content to a sub-template (for instance the content of a modal dialog), you can use the [custom block helper](#block-helpers) to provide a block of content. If you need more flexibility, typically just providing a named template a data context is the way to go. The sub-template can then just render that template with
+
+```html
+{{> Template.dynamic templateName dataContext}}
+```
+
+This is more or less the way that the [`kadira:blaze-layout`](https://atmospherejs.com/kadira/blaze-layout) package works in practice.
+
+<h3 id="pass-callbacks">Pass callbacks</h3>
+If you need to communicate *up* the template hierarchy, it's best to pass a *callback* for the subtemplate to call.
+
+For instance, only one todo item can be currently in the editing state at a time, so the `listsShow` template manages the state of which is edited. So when you focus on an item, that item needs to tell the list's template to make it the "edited" one. To do that, we pass a callback into the `todosItem` template, and it calls it:
+
+```html
+{{> todosItem (todoArgs todo)}}
+```
+
+```js
+Template.listsShow.helpers({
+  todoArgs(todo) {
+    const instance = Template.instance();
+    return {
+      todo,
+      editing: instance.state.equals('editingTodo', todo._id),
+      onEdit(doEdit) {
+        instance.state.set('editingTodo', doEdit ? todo._id : false);
+      }
+    };
+  }
+});
+
+Template.todosItem.events({
+  'focus input[type=text]'() {
+    this.onEdit(true);
```

Comment:
Should this be `Template.instance().data.onEdit`? If you use any of the data context changing block helpers it won't work anymore.